### PR TITLE
Add comment classification and soft-deletes

### PR DESCRIPTION
**Pull request by jbn**
This is a big commit that should have been done in stages but was not. It has two major changes,

1. It adds soft deletes for comments instead of hard ones;
2. It adds a comment classification job to get a real-time view of behavior

(1) is an easy win by just using the built-in soft-deletes provided by SQLBoiler.

To emphasis, (2) does no do auto-moderation. It only classifies comments for human-in-the-loop interventions. It also still lacks one of the most critical categories: doxxing.

In addition to these major changes it adds,

- A small docker script for testing against an ephemeral mysql instance
- Adds an execution guard to the OICD initialization so developers can work on the code without an internet connection without panics, set by `SKIP_OICD=1`

### CODE REVIEW

File: jobs/jobs.go [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package jobs
 
 import (
+	"github.com/OdyseeTeam/commentron/jobs/commentclassification"
```
Comment (nikooo777) [inline jobs/jobs.go:4]: settings > tools > file watchers > goimports

so imports are correctly sorted

**Review by nikooo777 — CHANGES_REQUESTED** [review]
a couple of changes, otherwise good work!

File: jobs/jobs.go (hunk 1 above)
Comment (jbn) [reply jobs/jobs.go]: fixed

File: jobs/commentclassification/package.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	"net/http"
+	"strings"
+	"sync"
+)
+
+var (
+	commentClassificationInProgress bool
+	commentClassificationMutex      sync.Mutex
+	defaultBatchSize                = 10
+	inferenceServiceURI             = "http://localhost:2345/api/v1/wtf"
```
[the remark this answers is not in the document]
Comment (jbn) [reply jobs/commentclassification/package.go:17]: kinda...(see DMs)

File: jobs/commentclassification/content_moderation.go [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+	return comments, nil
+}
+
+// Get the last known timestamp for the comment Classifications.
+//
+// WARNING: this assumes the `created_at` column replicates the `timestamp`.
+// WARNING: depending on timestamp granularity, this can miss some comments?
+func getLastKnownClassificationTimestamp() (int, error) {
+	cc, err := model.CommentClassifications(qm.OrderBy("timestamp DESC")).One(db.RO)
```
[the remark this answers is not in the document]
Comment (jbn) [reply jobs/commentclassification/content_moderation.go]: Oh wow, nice!

File: jobs/commentclassification/content_moderation.go [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+		Timeout: time.Second * 20,
+	}
+	resp, err := client.Post(inferenceServiceURI, "application/json", bytes.NewReader(reqBytes))
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse it
+	var classificationResp classificationResp
+	err = json.NewDecoder(resp.Body).Decode(&classificationResp)
```
[the remark this answers is not in the document]
Comment (jbn) [reply jobs/commentclassification/content_moderation.go:119]: Fixed

File: jobs/commentclassification/content_moderation.go [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+	client := http.Client{
+		Timeout: time.Second * 20,
+	}
+	resp, err := client.Post(inferenceServiceURI, "application/json", bytes.NewReader(reqBytes))
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse it
+	var classificationResp classificationResp
```
[the remark this answers is not in the document]
Comment (jbn) [reply jobs/commentclassification/content_moderation.go:118]: Fixed

File: README.md [hunk 6]
```diff
@@ -27,6 +27,9 @@ in the `.env` file locally with the username and password.
 
 Then once mysql is running you can hit play. 
 
+If you are lazy and have docker installed, you can use the `./scripts/test_db_via_docker.sh` script
+to start a mysql 5.7 container with appropriate default users, tables, and privilages for testing.
```
[the remark this answers is not in the document]
Comment (jbn) [reply README.md]: Updated my new script and README explanation, as well as old README section for 5.7.

**Pull request merged**